### PR TITLE
Add support for weaver-test

### PR DESCRIPTION
**Pull request by dabd**
Add support for https://github.com/disneystreaming/weaver-test.

### CODE REVIEW

File: generated-docs/out/usage/modifying.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ with the `Diff.ignore` instance.
 ## collection support
 
 Specify how objects within particular collection within particular diff instance should be matched.
-We distinguish free main types of collections:
+We distinguish three main types of collections:
```
Comment (ghostbuster91) [inline generated-docs/out/usage/modifying.md:45]: thanks :sweat_smile:

**Review by ghostbuster91 — COMMENTED** [review]
Look good, thanks for the great work. Let me fix the CI and then we will rebase your PR.

**Comment by ghostbuster91** [comment]
CI works again, please rebase your PR.

**Comment by ghostbuster91** [comment]
Seems like we need to update scala3 version to 3.2.2 to have waver compile.

**Comment by dabd** [comment]
> Seems like we need to update scala3 version to 3.2.2 to have waver compile.

Perhaps the scala3 upgrade should be done in another PR since the minor version bump might not be trivial, then I will rebase?

File: build.sbt [hunk 2]
```diff
@@ -96,7 +96,7 @@ lazy val core = (projectMatrix in file("core"))
   )
   .jvmPlatform(
     scalaVersions = List(scala212, scala213, scala3),
-      settings = enableMimaSettings
+    settings = enableMimaSettings
```
Comment (dabd) [inline build.sbt:99]: scalafmt.

Comment (ghostbuster91) [reply build.sbt:99]: I should've added it as a CI check.

**Comment by ghostbuster91** [comment]
Released as 0.9.0, thanks again for the contribution!

**Comment by dabd** [comment]
> Released as 0.9.0, thanks again for the contribution!

Thank you so much, I realise now I forgot to bump the versions matrix to 0.9.0 in weaver.md, due to the scala upgrade.

**Comment by ghostbuster91** [comment]
Ah indeed, no worries I will fix that.

**Comment by ghostbuster91** [comment]
If you wonder why the documentation does not reflect your recent changes it is because there were some breaking changes in readthedocs and our documentation build is failing :upside_down_face:

**Comment by ghostbuster91** [comment]
@dabd fixed :)

**Pull request merged**